From: SeongJae Park <sj@kernel.org>
Date: Mon, 17 Feb 2025 12:48:16 -0800
Subject: [PATCH] mm/damon: add damos_filter fields for default allow/reject
 behavior

It is more intuitive to decide the default filtering stage allowance
behavior depending on the last filter's behavior.  Core layer filters
stage should keep the default behavior as allowing regardless of the
behavior of the last core layer-handled filter though, if there are any
operations layer handled filters, though.

Hence it is simpler to keep separate decision for the two categories.
Add two fields for the two store the behaviors.

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 include/linux/damon.h | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/include/linux/damon.h b/include/linux/damon.h
index 7a4cf37e8b48..689a064726f9 100644
--- a/include/linux/damon.h
+++ b/include/linux/damon.h
@@ -497,6 +497,9 @@ struct damos {
 	unsigned long next_apply_sis;
 	/* informs if ongoing DAMOS walk for this scheme is finished */
 	bool walk_completed;
+	/* whether to reject core/ops filters umatched regions */
+	bool core_filters_default_reject;
+	bool ops_filters_default_reject;
 /* public: */
 	struct damos_quota quota;
 	struct damos_watermarks wmarks;
-- 
2.39.5

